projects/common-form-elements/src/lib/dynamic-dropdown/dynamic-dropdown.component.ts

```diff
@@ -1,4 +1,4 @@
-import {Component, Input, OnChanges, OnDestroy, OnInit, Output, SimpleChanges, EventEmitter} from '@angular/core';
+import {Component, Input, OnChanges, OnDestroy, OnInit, Output, SimpleChanges, EventEmitter, HostListener} from '@angular/core';
```

Comment:
In production, we're facing an [issue ](https://project-sunbird.atlassian.net/browse/SB-26881?focusedCommentId=87664&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-87664 )with the performance of this component. The most probable cause is a memory leak when the list of elements in the existing `ngFor` is too many. Can you try replicating that issue and see if the second strategy mentioned [here](https://blog.angular.io/3-tips-for-angular-runtime-performance-from-the-real-world-d467fbc8f66e) can fix it?